content/reference/wadm/model.md

```diff
@@ -1,8 +1,111 @@
+
+When application specifications are stored, they are keyed by name and a history of all versions is maintained, with the _most recently pushed_ version being considered the newest. We do not make assumptions about whether one version string (e.g. is `v1.10` is newer than `v1.1`?) is newer or older than any other. Version history is based purely on time of storage.
+
+## Components
+
+While OAM allows us to define any component in a specification, there are only a few components with which wadm is concerned:
+* `actor` - represents a specification of an actor
+* `capability` - represents a specification of a capability provider
+
+Within the `components` field of a specification, you define an `actor` as follows:
+
+```yaml
+spec:
+  components:
+    - name: userinfo
+      type: actor
+      properties:
+        image: wasmcloud.azurecr.io/fake:1
+      traits:
+        ...
+```
+
+The `image` property of the `actor` component contains an OCI (or bindle) image reference URL. When you attempt to store this model, wadm will reach out to the artifact repository and attempt to pull some metadata about that image (such as its primary key, embedded security information, etc). This means that storing application specifications with invalid/unreachable OCI references is not allowed.
+
+To define a capability provider, we include a `capability` component, as follows:
+
+```yaml
+    - name: webcap
+      type: capability
+      properties:
+        contract: wasmcloud:httpserver
+        image: wasmcloud.azurecr.io/httpserver:0.13.1
+        link_name: default
+```
+
+Just like when manipulating a lattice _imperatively_, the things that differentiate one capability provider from another are its contract, its public key (which we obtain by looking up the `image`), and its link name.
+
+## Traits
+Traits are, as their name applies, metadata associated with a `component`. The OAM trait system is completely extensible, so as wadm gains more functionality, it can support more traits. Right now, the following traits are supported:
+
+* `spreadscaler`
+* `linkdef`
+
+### Spread Scaler
+The `spreadscaler` trait contains a specification for how you would like to scale a set number of instances of an actor. We call it a _spread_ scaler because you declare how you would like the instances of that actor spread across the hosts within your lattice by specifying targets with host _labels_. You can think of this like affinity and anti-affinity rules combined with a scale specification.
+
+Take a look at the following sample `spreadscaler` spec:
+
+```yaml
+      traits:
+        - type: spreadscaler
+          properties:
+            replicas: 4
+            spread:
+              - name: eastcoast
+                requirements:
+                  zone: us-east-1
+                weight: 80
+              - name: westcoast
+                requirements:
+                  zone: us-west-1
+                weight: 20
+```
```

Comment:
Could you include a simple example of just specifying the number of replicas after https://github.com/wasmCloud/wadm/pull/35 to show the simplest case?